compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/DerivedState.kt

```diff
@@ -64,8 +64,10 @@ private class DerivedSnapshotState<T>(
     private var first: ResultRecord<T> = ResultRecord()
 
     private class ResultRecord<T> : StateRecord() {
+        private object Unset
```

Comment:
You can save on APK size / class-loaded memory by changing this to just an arbitrary instance instead.
```suggestion
        companion object {
            private val unset = Any()
        }
```
(also needs additional refactors of the code below due to casing change)

I'm not sure why the `Lazy` implementation feels the need to use an `object` which incurs the penalty of a whole class in the binary, a class being loaded at runtime, and a static field to forever retain the instance.

Now you might say "but the `companion object` does the very same thing!". However, R8 will eliminate the companion and hoist the reference to a static on the bytecode class for `ResultRecord` whereas it cannot do this for `Unset` as you might be relying on the fact that `Unset::class != Any::class` somewhere.

It seems small, but this kind of stuff can really be death by a thousand cuts...

Otherwise 👍

Reply:
Makes sense, thanks!
I kept the casing as `Unset` to match other Compose top-level properties.